File: mi_conjunto_guardas/lib/models/receipt.dart
import 'public_company.dart';

/// Modelo de 'recibo'
class Receipt {
  int receiptId;
  String receptionDate;
  String deliveryDate;
  String notes;
  bool checked;
  PublicCompany publicCompany;

  Receipt(
      {this.receiptId,
      this.receptionDate,
      this.deliveryDate,
      this.notes,
      this.checked,
      this.publicCompany});

  factory Receipt.fromJson(Map<String, dynamic> json) {
    return Receipt(
        receiptId: json["receiptId"],
        receptionDate: json["receptionDate"],
        deliveryDate: json["deliveryDate"],
        notes: json["notes"],
        checked: false,
        publicCompany: PublicCompany.fromJson(json["PublicCompany"]));
  }
}

File: server/sequelize/models/Receipt.js
import DataTypes from "sequelize";

/**
 * Modelo de 'recibo' en base de datos
 * @param {*} sequelize 
 */
export const receiptModel = (sequelize) => {
  sequelize.define(
    "Receipt",
    {
      receiptId: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        allowNull: false,
        autoIncrement: true,
        field: "id_recibo",
      },
      receptionDate: {
        type: DataTypes.DATE,
        allowNull: false,
        field: "fecha_recepcion",
      },
      deliveryDate: {
        type: DataTypes.DATE,
        field: "fecha_entrega",
      },
      notes: {
        type: DataTypes.STRING,
        field: "observaciones",
      },
    },
    {
      tableName: "recibo",
      timestamps: false,
    }
  );
};
